### PR TITLE
Replace custom webpack integration with webpacker

### PR DESCRIPTION
**Pull request by jhawthorn**
What does it do?
----------------
Adds webpacker and moves javascripts under webpacker

How does it work?
-----------------
Moves the javascripts from `app/webpack` into `app/javascript`, the new canonical location for webpack.

Webpacker will compile a "pack" for each file in the `app/javascript/packs` directory and places them in `public/packs/` without going through sprockets. Those packs can then be linked to using `javascript_pack_tag` (there's a `public/packs/digests.json` which powers this).

The new commands for invoking webpack are:
`bin/webpack`
`bin/webpack-watcher`
`bin/webpack-dev-server` (which has some related configuration in config/environments/development.rb)

Anything to consider when deploying?
------------------------------------
Yeah, it might not work, but I think it will. The `yarn run compile` commands now just call `bin/webpack`. `rake assets:precompile` also will implicitly run `rake webpacker:compile`.

Screenshots (Optional)
----------------------

### CODE REVIEW

File: config/webpack/shared.js [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+
+  module: {
+    rules: [
+      {
+        test: /\.js(.erb)?$/,
+        exclude: /node_modules/,
+        loader: 'babel-loader',
+        options: {
+          presets: [
+            [ 'latest', { 'es2015': { 'modules': false } } ]
```
Comment (jarednorman) [inline config/webpack/shared.js]: Doesn't look like you've removed the `.babelrc` which specifies different options than this. Probably need to axe it and move that stuff here? Or if this config respects the `.babelrc`, move those options there? Or explain what's goin' on here so I feel better.

**Review by jarednorman — CHANGES_REQUESTED** [review]
This looks all fine except I'm wary of the Babel config change.

File: config/webpack/shared.js (hunk 1 above)
Comment (jhawthorn) [reply config/webpack/shared.js]: I have moved the options in here.

**Review by jarednorman — APPROVED** [review]
Go for it.

**Comment by jhawthorn** [comment]
Going for it

**Comment by jhawthorn** [comment]
Gone

**Comment by jhawthorn** [comment]
Update: [went](https://travis-ci.org/PorkChopClub/porkchop/builds/205449399)

**Pull request merged**